prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -460,6 +460,13 @@ class ReportFunction : Logging {
                     validatedRequest.routeTo,
                     validatedRequest.warnings,
                 )
+                .filter { (report, receiver) ->
+                    if (report.itemCount > 0) {
+                        return@filter true
+                    }
```

Comment:
I would change it to an if/else, so you do not have to force the returns.

Reply:
I prefer early returns as a style, the code is more delete-able, but agreed it's funky with the named return

Reply:
This made me reconsider the pattern completely, I think I'm happier with what came out it's very clear what it's doing and why.